common/Makefile

```diff
@@ -0,0 +1,76 @@
+BUILDDIR := ${current_dir}/build
+BOARD_BUILDDIR := ${BUILDDIR}/${TARGET}
+
+# Set board properties based on TARGET variable
+ifeq ($(TARGET),arty_35)
+  DEVICE := xc7a50t_test
+  BITSTREAM_DEVICE := artix7
+  PARTNAME := xc7a35tcsg324-1
+else ifeq ($(TARGET),arty_100)
+  DEVICE := xc7a100t_test
+  BITSTREAM_DEVICE := artix7
+  PARTNAME := xc7a100tcsg324-1
+else ifeq ($(TARGET),nexys4ddr)
+  DEVICE := xc7a100t_test
+  BITSTREAM_DEVICE := artix7
+  PARTNAME := xc7a100tcsg324-1
+else ifeq ($(TARGET),zybo)
+  DEVICE := xc7z010_test
+  BITSTREAM_DEVICE := zynq7
+  PARTNAME := xc7z010clg400-1
+else ifeq ($(TARGET),nexys_video)
+  DEVICE := xc7a200t_test
+  BITSTREAM_DEVICE := artix7
+  PARTNAME := xc7a200tsbg484-1
+else ifeq ($(TARGET),basys3)
+  DEVICE := xc7a50t_test
+  BITSTREAM_DEVICE := artix7
+  PARTNAME := xc7a35tcpg236-1
+else
+  $(error Unsupported board type)
+endif
+
+# Determine the type of constraint being used
+ifneq (${XDC},)
+  XDC_CMD := -x ${XDC}
+endif
+ifneq (${SDC},)
+  SDC_CMD := -s ${SDC}
+endif
+ifneq (${PCF},)
+  PCF_CMD := -p ${PCF}
+endif
+
+.DELETE_ON_ERROR:
+
+# Build design
+all: ${BOARD_BUILDDIR}/${TOP}.bit
+
+${BOARD_BUILDDIR}:
+	mkdir -p ${BOARD_BUILDDIR}
+  #Check if design includes initialization files for memory
+  ifneq (${MEM_INIT},)
+    #if there are memory initialization files link them to the build/board_type directory
+	  ln -s ${MEM_INIT} ${BOARD_BUILDDIR}
+  endif
```

Comment:
I think that creating symlinks of the mem init files can be avoided. 

I see this is done as the init files are present for the linux litex demo, but I am quite sure that the symlinking step was actually unnecessary.